src/common/actions/register-name.js

```diff
@@ -198,3 +211,79 @@ export function registerNameClear(id) {
     }
   };
 }
+
+export async function internalNameOwnership(root, discharge, snapName) {
+  if (!snapName) {
+    throw new APICompatibleError({
```

Comment:
I guess you're right. Especially that it's most likely to be an error caused by developer mistake rather then application state.